yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -598,8 +598,16 @@ private[spark] class Client(
     ).foreach { case (flist, resType, addToClasspath) =>
       flist.foreach { file =>
         val (_, localizedPath) = distribute(file, resType = resType)
-        if (addToClasspath && localizedPath != null) {
-          cachedSecondaryJarLinks += localizedPath
+        if (addToClasspath) {
+          if (localizedPath != null) {
+            cachedSecondaryJarLinks += localizedPath
+          }
+        } else {
+          if (localizedPath != null) {
+            throw new IllegalArgumentException(s"Attempt to add ($file) multiple times. " +
+              "Please check the values of 'spark.yarn.dist.files' and/or " +
```

Comment:
can you remove the part about check values of those specific configs because there are multiple ways for these to specified (configs or --files, --jars, etc).   Perhaps just say please check the values you specified for uploading files,jars, and archives to make sure one isn't specified multiple times..